python/lsst/daf/persistence/storage.py

```diff
@@ -0,0 +1,124 @@
+        ----------
+        scheme : str
+            Name of the `scheme` the class is being registered for, which appears at the beginning of a URI.
+        cls : class object
+            A class object that should be used for a given scheme.
+        """
+        if scheme in Storage.storages:
+            raise RuntimeError("Scheme '%s' already registered:%s" % (scheme, Storage.storages[scheme]))
+        Storage.storages[scheme] = cls
+
+    @staticmethod
+    def getRepositoryCfg(uri):
+        """Get a RepositoryCfg from a location specified by uri."""
+        ret = None
+        parseRes = urlparse.urlparse(uri)
+        if parseRes.scheme in Storage.storages:
+            ret = Storage.storages[parseRes.scheme].getRepositoryCfg(uri)
+        else:
+            raise RuntimeError("No storage registered for scheme %s" % parseRes.scheme)
+        return ret
+
+    @staticmethod
+    def putRepositoryCfg(cfg, uri):
+        """Write a RepositoryCfg object to a location described by uri"""
+        ret = None
+        parseRes = urlparse.urlparse(uri)
+        if parseRes.scheme in Storage.storages:
+            ret = Storage.storages[parseRes.scheme].putRepositoryCfg(cfg, uri)
+        else:
+            raise RuntimeError("No storage registered for scheme %s" % parseRes.scheme)
+        return ret
+
+
+    @staticmethod
+    def getMapperClass(uri):
+        """Get a mapper class cfg value from location described by uri.
+
+        Note that in legacy repositories the mapper may be specified by a file called _mapper at the uri 
+        location, and in newer repositories the mapper would be specified by a RepositoryCfg stored at the uri
+        location.
+
+        .. warning::
+
+            Storage is 'wet paint' and very likely to change during factorization of Butler back end and 
+            storage formats (DM-6225). Use of it in production code other than via the 'old butler' API is 
+            strongly discouraged.
+
+        """
+        ret = None
+        parseRes = urlparse.urlparse(uri)
+        if parseRes.scheme in Storage.storages:
+            ret = Storage.storages[parseRes.scheme].getMapperClass(uri)
+        else:
+            raise RuntimeError("No storage registered for scheme %s" % parseRes.scheme)
+        return ret
+
+
+    @staticmethod
+    def makeFromURI(uri):
+        '''Instantiate a storage sublcass from a URI.
+
+        .. warning::
+
+            Storage is 'wet paint' and very likely to change during factorization of Butler back end and 
+            storage formats (DM-6225). Use of it in production code other than via the 'old butler' API is 
+            strongly discouraged.
+
+        :param uri: the uri to a locaiton that contains a repositoryCfg.
+        :return: a Storage subclass instance.
+        '''
+        ret = None
+        parseRes = urlparse.urlparse(uri)
+        if parseRes.scheme in Storage.storages:
+            theClass = Storage.storages[parseRes.scheme]
```

Comment:
Shouldn't matter in this case, but as a good general practice I'd recommend instead doing
```
theClass = Storage.storages.get(parseRes.scheme, None)
if theClass is None:
    raise ...
```
just so there's only one dict lookup.